code_generation/producer.py

```diff
@@ -142,126 +161,200 @@ def writecalls(self, config):
+    "RequireObjects",
     'physicsobject::FilterObjects({df}, "{require_candidate}", {require_candidate_number}, "{require_candidate}")',
     [],
     None,
+    ["global"],
     ["require_candidate", "require_candidate_number"],
 )
 
 MTPairSelection = Producer(
+    "MTPairSelection",
     'pairselection::mutau::PairSelection({df}, {input_coll}, "{output}")',
     [q.Tau_pt, q.Tau_IDraw, q.Muon_pt, q.Muon_iso, q.good_taus_mask, q.good_muons_mask],
     q.ditaupair,
+    ["mt"],
 )
 
 GoodMTPairFilter = Producer(
+    "GoodMTPairFilter",
     'pairselection::filterGoodPairs({df}, "{input}", "GoodMuTauPairs")',
     [q.ditaupair],
     None,
+    ["mt"],
 )
 
 LVMu1 = Producer(
+    "LVMu1",
     'lorentzvectors::build({df}, {input_coll}, 0, "{output}")',
     [q.ditaupair, q.Muon_pt, q.Muon_eta, q.Muon_phi, q.Muon_mass],
     q.p4_1,
+    ["mt"],
 )
 LVMu2 = Producer(
+    "LVMu2",
     'lorentzvectors::build({df}, {input_coll}, 1, "{output}")',
     [q.ditaupair, q.Muon_pt, q.Muon_eta, q.Muon_phi, q.Muon_mass],
     q.p4_2,
+    ["mt"],
 )
 LVTau1 = Producer(
+    "LVTau1",
     'lorentzvectors::build({df}, {input_coll}, 0, "{output}")',
     [q.ditaupair, q.Tau_pt, q.Tau_eta, q.Tau_phi, q.Tau_mass],
     q.p4_1,
+    ["mt"],
 )
 LVTau2 = Producer(
+    "LVTau2",
     'lorentzvectors::build({df}, {input_coll}, 1, "{output}")',
     [q.ditaupair, q.Tau_pt, q.Tau_eta, q.Tau_phi, q.Tau_mass],
     q.p4_2,
+    ["mt"],
 )
 
-pt_1 = Producer('quantities::pt({df}, varSet, "{output}", "{input}")', [q.p4_1], q.pt_1)
-pt_2 = Producer('quantities::pt({df}, varSet, "{output}", "{input}")', [q.p4_2], q.pt_2)
+pt_1 = Producer(
+    "pt_1",
+    'quantities::pt({df}, varSet, "{output}", "{input}")',
+    [q.p4_1],
+    q.pt_1,
+    ["mt"],
+)
+pt_2 = Producer(
+    "pt_2",
+    'quantities::pt({df}, varSet, "{output}", "{input}")',
+    [q.p4_2],
+    q.pt_2,
+    ["mt"],
+)
 eta_1 = Producer(
-    'quantities::eta({df}, varSet, "{output}", "{input}")', [q.p4_1], q.eta_1
+    "eta_1",
+    'quantities::eta({df}, varSet, "{output}", "{input}")',
+    [q.p4_1],
+    q.eta_1,
+    ["mt"],
 )
 eta_2 = Producer(
-    'quantities::eta({df}, varSet, "{output}", "{input}")', [q.p4_2], q.eta_2
+    "eta_2",
+    'quantities::eta({df}, varSet, "{output}", "{input}")',
+    [q.p4_2],
+    q.eta_2,
+    ["mt"],
 )
 phi_1 = Producer(
-    'quantities::phi({df}, varSet, "{output}", "{input}")', [q.p4_1], q.phi_1
+    "phi_1",
+    'quantities::phi({df}, varSet, "{output}", "{input}")',
+    [q.p4_1],
+    q.phi_1,
+    ["mt"],
 )
 phi_2 = Producer(
-    'quantities::phi({df}, varSet, "{output}", "{input}")', [q.p4_2], q.phi_2
+    "phi_2",
+    'quantities::phi({df}, varSet, "{output}", "{input}")',
+    [q.p4_2],
+    q.phi_2,
+    ["mt"],
 )
-UnrollLV1 = ProducerGroup(None, None, None, [pt_1, eta_1, phi_1])
-UnrollLV2 = ProducerGroup(None, None, None, [pt_2, eta_2, phi_2])
+UnrollLV1 = ProducerGroup("UnrollLV1", None, None, None, ["mt"], [pt_1, eta_1, phi_1])
```

Comment:
None, None None looks a bit stange here, either we add the keywords, or maybe change the ordering of the paramteters. From just looking at this line, its not clear at all what these Nones do

Reply:
Right, you suggested to add the keywords once before. That's probably a good thing and I will add it.